### PR TITLE
first part of task is done

### PR DESCRIPTION
**Pull request by andromash**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        return Storage.products.stream()
+                .filter(n -> n.getId().equals(product.getId()))
+                .map(p -> p = product)
+                .findFirst()
+                .orElseThrow();
```
Comment (aleksandr-hrankin) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: Чисто теоретически может быть ситуация что по одному id он найдет несколько объектов. Тогда нужно изменить их все. Можно через IntStream и методом forEach

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.products.remove(get(id).orElseThrow());
```
Comment (aleksandr-hrankin) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: Тоже может быть такая ситуация, прочитай за метод removeIf() - огонь метод!

File: src/main/java/com/internet/shop/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(List<Product> products, Long userId) {
+        this.products = products;
+        this.userId = userId;
+    }
```
Comment (aleksandr-hrankin) [inline src/main/java/com/internet/shop/model/Order.java:14]: Не знаю на сколько это верно, но я сразу в конструкторе инициализировал List, не передавая его

Comment (andromash) [reply src/main/java/com/internet/shop/model/Order.java:14]: We pass a list of products that we defined in the shopping cart. So we can't just say new ArrayList or something else;

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 1 above)
Comment (andromash) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: .map() will change all of them. We just return one

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 2 above)
Comment (andromash) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: agree here 👍

**Review by aleksandr-hrankin — APPROVED** [review]

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: @aleksandr-hrankin Id it's a unique  value of object

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: agree with using removeIf() but id should be unique for each object of some type

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(product.getId())) {
+                return Storage.products.set(i, product);
+            }
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: let's use stream api for this with the same approach

File: src/main/java/com/internet/shop/db/Storage.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public static final List<Order> orders = new ArrayList<>();
+    public static final List<ShoppingCart> shoppingCarts = new ArrayList<>();
+    public static final List<User> users = new ArrayList<>();
+    private static Long productId = 0L;
+    private static Long orderId = 0L;
+    private static Long shoppingCartId = 0L;
+    private static Long userId = 0L;
+
+    public static void addProduct(Product product) {
+        product.setId(++productId);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/db/Storage.java:21]:
```suggestion
        product.setId(productId++);
```
it's okay to have the first id = 0;

Comment (andromash) [reply src/main/java/com/internet/shop/db/Storage.java:21]: We decided that it is not on today's lecture

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 4 above)
Comment (andromash) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java:37]: i did before, but stream API is not changing anything in the storage. you can see it in previous commits

File: src/main/java/com/internet/shop/model/Order.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(List<Product> products, Long userId) {
+        this.products = products;
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/model/Order.java]: On QnA says should be new ArrayList() in the constructor.

File: src/main/java/com/internet/shop/model/Product.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+package com.internet.shop.model;
+
+public class Product {
+    private Long id;
+    private String name;
+    private Double price;
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/model/Product.java]: Maybe money should be in BigDecimal?

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public ShoppingCart(List<Product> products, Long userId) {
+        this.products = products;
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/model/ShoppingCart.java]: new...

File: src/main/java/com/internet/shop/model/Product.java (hunk 7 above)
Comment (andromash) [reply src/main/java/com/internet/shop/model/Product.java]: agree, it would be better :)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**